### PR TITLE
AiRanged refactors

### PR DESCRIPTION
**Pull request by k-bar**
Refactor AiRanged changes.

### CODE REVIEW

**Comment by qndel** [comment]
if you don't want these commits to get squashed, how about fixing that typo `ger` -> `get`? in commit description? :D

File: Source/monster.cpp [hunk 1]
```diff
@@ -2070,7 +2070,42 @@ void AiAvoidance(int i, bool special)
 	monster.CheckStandAnimationIsLoaded(md);
 }
 
-void AiRanged(int i, missile_id missileType, bool special)
+ missile_id GetMissileType(_mai_id ai)
+{
+	switch (ai) {
+	case AI_GOATMC:
+		return MIS_ARROW;
+		break;
```
Comment (qndel) [inline Source/monster.cpp:2078]: you don't need breaks after return

Comment (k-bar) [reply Source/monster.cpp:2078]: isn't that good practice or something ? :D

Comment (glebm) [reply Source/monster.cpp:2078]: Not after a return

**Comment by DakkJaniels** [comment]
Are these kind of changes being playtested before merging? Looks like this might be causing Laz's helpers to fire arrows instead of bloodstars because their AI is AI_LAZHELP, not AI_SUCC

**Comment by k-bar** [comment]
> Are these kind of changes being playtested before merging? Looks like this might be causing Laz's helpers to fire arrows instead of bloodstars because their AI is AI_LAZHELP, not AI_SUCC

yes, but unfortunately not through all kind of monsters.

**Pull request merged**